vmm/modules/security_info/security_info.c

```diff
@@ -107,19 +108,28 @@ void *mov_secinfo_to_internal(void *src)
 	return handle;
 }
 
-uint32_t mov_secinfo_from_internal(void *dest, void *handle)
+boolean_t mov_secinfo_from_internal(guest_cpu_handle_t gcpu,
+		uint64_t dst_gva,
+		uint64_t src_hva)
 {
-	uint32_t dev_sec_info_size;
+	uint32_t size;
+	boolean_t ret;
+	pf_info_t pfinfo;
+
+	D(VMM_ASSERT_EX(src_hva, "source address can't be NULL!\n"));
 
-	D(VMM_ASSERT_EX(handle, "dest and handle can't be NULL!\n"));
+	size = *(uint32_t *)src_hva;
 
-	dev_sec_info_size = _mov_info(dest, handle);
+	ret = gcpu_copy_to_gva(gcpu, dst_gva, src_hva, size, &pfinfo);
 
-	mem_free(handle);
+	memset((void *)src_hva, 0, size);
```

Comment:
add barrier(); after erase sensitive data.

Reply:
Get it. I will fix it in next patch

Reply:
Done